### PR TITLE
Pagination

### LINKED ISSUE #9: Implement Jedis Connection Pooling
In the current RedisAccessObject implementation, connections to Redis are not managed centrally. In order to ensure we aren't wasting resources, a common (singleton) connection pool should be implemented. This could be done in a single class.

### LINKED ISSUE #28: Implement Pagination on back-end
`TopConentController`, `HashtagContentController`, and `TrendingContentController` endpoints should be edited to support pagination.

### LINKED ISSUE #29: Implement Pagination on front-end
Each front-end page (hashtag, top, trending) should poll successive pages of their respective endpoint, ideally using lazy loading.

This is blocked by #28

### LINKED ISSUE #36: Implement Sorting by Hashtag

### LINKED ISSUE #37: Implement Sorting by Trending

### LINKED ISSUE #38: Implement Top Hashtag functionality in backend
This is threefold:

- Need to support evaluating which top hashtags we have
- Need to support storing top hashtags
- Need to implement top hashtag controller (endpoint for front end)

### PR DESCRIPTION
**Pull request by MariaStrasky**
Resolves #9 - Redis connection pooling now fully functional and thread safe
Resolves #28 
Resolves #29 - Pagination now working on front & back end
Resolves #36 - Sorting by hashtag now available, endpoint fully supports pagination
Resolves #37 - Sorting by trending now available; may need some work based on how we gather & display data
Resolves #38 - Top Hashtags now evaluated by sorting node, stored in data tier, and available for query on the TopHashtags endpoint.

### CODE REVIEW

File: app/views/main.scala.html
Comment (edolinsky) [inline app/views/main.scala.html:55]: Is this still supposed to route to testing? Or resolved in ImproveUI branch?

File: app/Module.java
Comment (reoliveira) [inline app/Module.java:27]: +1

File: app/views/main.scala.html
Comment (reoliveira) [reply app/views/main.scala.html:55]: ^ same question. Whatever is going to master should be prod ready

Comment (MariaStrasky) [reply app/views/main.scala.html:55]: We don't have the search bar implemented in this version, so there's no way for users to pass in which hashtag they want to look for, so making it prod ready will have to wait until Improve UI gets merged into master, and this one gets merged in, then we can make a new branch for implementing the hashtag front-end

**Review by edolinsky — APPROVED** [review]
Reviewed as per Reid

**Pull request merged**
